src/components/BoardColor/BoardColor.tsx

```diff
@@ -0,0 +1,291 @@
+import { useState } from 'react';
+import {
+    Button,
+    ColorPicker,
+    ColorSlider,
+    ColorSwatch,
+    ColorThumb,
+    Dialog,
+    DialogTrigger,
+    Label,
+    ListBox,
+    ListBoxItem,
+    Popover,
+    parseColor,
+    Selection,
+    SliderOutput,
+    SliderTrack,
+} from 'react-aria-components';
+import './ReactAria.css'
+import { BoardColorObj } from "../../types/chess";
+
+interface TemplateColorProps {
+  id: string,
+  lightColor: string,
+  darkColor: string,
+}
+
+const TemplateColor: React.FC<TemplateColorProps> = ({ id, lightColor, darkColor }) => {
+  return (
+    <ListBoxItem id={id} textValue={id} className='react-aria-ColorSwatchPickerItem'>
+      <ColorSwatch color={parseColor(lightColor)}/>
+      <ColorSwatch color={parseColor(darkColor)}/>
+    </ListBoxItem>
+  )
+}
+
+interface BoardColorProps {
+    onChangeColor: (selectedColor: BoardColorObj) => void;
+    selectedColor: BoardColorObj;
+}
+
+const BoardColor: React.FC<BoardColorProps> = ({ onChangeColor, selectedColor }) => {
+    // Color Picker is set to Amber by default
+    const [hueColor, setHueColor] = useState(parseColor('hsl(25, 100%, 50%)'));
+    const [darkSlider, setDarkSlider] = useState(parseColor(selectedColor.dark));
+    const [lightSlider, setLightSlider] = useState(parseColor(selectedColor.light));
+    const [darkBoardColor, setDarkBoardColor] = useState(parseColor(selectedColor.dark));
+    const [lightBoardColor, setLightBoardColor] = useState(parseColor(selectedColor.light));
+    const [selectedTemplateColor, setSelectedTemplateColor] = useState<Selection>(new Set(['amber']))
+    const [isPopoverOpen, setPopoverOpen] = useState(false);
+    // 0 index is the light color and 1 index is the dark color
+    const colorMap: Record<string, [string, string]> = {
+        amber: ["hsl(25, 100%, 89%)", "hsl(25, 100%, 36%)"],
+        gray: ["hsl(217, 21%, 96%)", "hsl(217, 21%, 27%)"],
+        blue: ["hsl(193, 100%, 90%)", "hsl(193, 100%, 29%)"],
+        green: ["hsl(86, 100%, 89%)", "hsl(86, 100%, 25%)"],
+        purple: ["hsl(295, 100%, 95%)", "hsl(295, 100%, 36%)"],
+    };
```

Comment:
@Trevbot50 these adobe components only work with hsl and rgba. The tailwindcss colors were in oklch so I did a rough conversion of them on this [site](https://folge.me/tools/oklch-color-picker). Also, due to the limitations of this implementation the saturation of the light/dark side has to be the same. Lmk if these colors aren't close enough to the og ones and we can maybe tune em